src/compose/ComposeMenu.js

```diff
@@ -84,90 +77,128 @@ export const chooseUploadImageFilename = (uri: string, fileName: string): string
-        showErrorAlert(_('Error'), errorMessage);
-        logging.error('Unexpected error from image picker', {
-          errorCode,
-          errorMessage: errorMessage ?? '[nullish]',
+            ],
+          );
+        } else if (errorCode === 'camera_unavailable') {
+          showErrorAlert(_('Error'), _('Camera unavailable.'));
+        } else {
+          const { errorMessage } = response;
+          showErrorAlert(_('Error'), errorMessage);
+          logging.error('Unexpected error from image picker', {
+            errorCode,
+            errorMessage: errorMessage ?? '[nullish]',
+          });
+        }
+        return;
+      }
+
+      const { assets } = response;
+
+      if (!assets || !assets[0]) {
+        // TODO: See if we these unexpected situations actually happen. …Ah,
+        //   yep, reportedly (and we've seen in Sentry):
+        //   https://github.com/react-native-image-picker/react-native-image-picker/issues/1945
+        showErrorAlert(_('Error'), _('Failed to attach your file.'));
+        logging.error('Image picker response gave falsy `assets` or falsy `assets[0]`', {
+          '!assets': !assets,
         });
+        return;
       }
-      return;
-    }
 
-    // TODO: support sending multiple files; see library's docs for how to
-    // let `assets` have more than one item in `response`.
-    const firstAsset = response.assets && response.assets[0];
+      const attachments = [];
+      let numMalformed = 0;
+      assets.forEach((asset, i) => {
+        const { uri, fileName } = asset;
 
-    const { uri, fileName } = firstAsset ?? {};
+        if (uri == null || fileName == null) {
+          // TODO: See if these unexpected situations actually happen.
+          logging.error('An asset returned from image picker had nullish `url` and/or `fileName`', {
+            'uri == null': uri == null,
+            'fileName == null': fileName == null,
+            i,
+          });
+          numMalformed++;
+          return;
+        }
 
-    if (!firstAsset || uri == null || fileName == null) {
-      // TODO: See if we these unexpected situations actually happen. …Ah,
-      //   yep, reportedly (and we've seen in Sentry):
-      //   https://github.com/react-native-image-picker/react-native-image-picker/issues/1945
-      showErrorAlert(_('Error'), _('Something went wrong, and your message was not sent.'));
-      logging.error('Unexpected response from image picker', {
-        '!firstAsset': !firstAsset,
-        'uri == null': uri == null,
-        'fileName == null': fileName == null,
+        attachments.push({ name: chooseUploadImageFilename(uri, fileName), url: uri });
       });
-      return;
-    }
 
-    dispatch(uploadFile(destinationNarrow, uri, chooseUploadImageFilename(uri, fileName)));
-  };
+      if (numMalformed > 0) {
+        let msg = undefined;
+        if (assets.length === 1 && numMalformed === 1) {
+          msg = _('Failed to attach your file.');
+          return;
```

Comment:
This sets a local and then returns -- so the assignment has no effect, as nothing else can see it.

Perhaps just put the `showErrorAlert` call here? It's pretty concise, not much more complex than the variable assignment.